MeesGame/AI/FloodFill.cs

```diff
@@ -0,0 +1,176 @@
+                // Using else if (action != PlayerAction.NONE) cause a bug somehow.
+                else if (action == PlayerAction.NORTH || action == PlayerAction.EAST || action == PlayerAction.SOUTH || action == PlayerAction.WEST)
+                {
+                    // Obstacles are checked earlier in Flood() so we only need to check whether the new point is in the tilefield.
+                    if (!tileField.OutOfTileField(p + action.ToDirection().ToPoint()))
+                    {
+                        Flood(p + action.ToDirection().ToPoint());
+                    }
+                }
+            }
+
+            if (tileField.GetTile(p.X, p.Y).TileType == TileType.End)
+            {
+                // If there is no best route yet.
+                if (bestRoute.Count == 0)
+                    bestRoute = new Stack<Point>(new Stack<Point>(currentRoute));
+                // If the current route is better than the curent best route.
+                else if (bestRoute.Count > currentRoute.Count)
+                    bestRoute = new Stack<Point>(new Stack<Point>(currentRoute));
+
+                return;
+            }
+            
+            // All directions are done, return to the previous point.
+            currentRoute.Pop();
+        }
+
+        /// <summary>
+        /// Resets the visited array.
+        /// </summary>
+        private void UnvisitEverything()
+        {
+            for (int i = 0; i < visited.GetLength(0); i++)
+            {
+                for (int j = 0; j < visited.GetLength(1); j++)
+                {
+                    visited[i, j] = false;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Converts a route to a path.
+        /// </summary>
+        /// <param name="route"> the route to be converted to a path</param>
+        private void RouteToPath(Stack<Point> route)
+        {
+            Point start = route.Pop();
+            while (route.Count > 1)
+            {
+                Point end = route.Pop();
+                path.Add(GetActionFromPoints(start, end));
+                start = end;
+            }
+        }
+
+        /// <summary>
+        /// Returns the action needed to get from start to end.
+        /// </summary>
+        /// <param name="start"> starting point </param>
+        /// <param name="end">ending point </param>
+        /// <returns> The action needed to get from start to end. </returns>
+        private PlayerAction GetActionFromPoints(Point start, Point end)
+        {
+            if (start + Direction.NORTH.ToPoint() == end)
+                return PlayerAction.NORTH;
+            else if (start + Direction.EAST.ToPoint() == end)
+                return PlayerAction.EAST;
+            else if (start + Direction.SOUTH.ToPoint() == end)
+                return PlayerAction.SOUTH;
+            else if (start + Direction.WEST.ToPoint() == end)
```

Comment:
Use 'if' instead of 'else if'. The return statement will terminate the function anyways.